plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -73,6 +73,9 @@ type predicateMetadata struct {
 	podRequest                *schedulercache.Resource
 	podPorts                  map[int]bool
 	matchingAntiAffinityTerms []matchingPodAntiAffinityTerm
+	// Service pods cache, otherwise this is a large query, will slow things down
+	// when there are 100s of services/pods.
+	servicePods []*api.Pod
```

Comment:
will dig into this and see , maybe I can update the unit tests to uncover that race